### PR TITLE
Auth: Add org to role mappings support to Gitlab integration

### LINKED ISSUE #88620: Auth: Org to role mapping for GitLab
Implement org to role mapping support for GitLab following the changes of the following PRs:

- https://github.com/grafana/grafana/pull/87394
- https://github.com/grafana/grafana/pull/88537

Probably org_mapping should only support the `groups` attribute of the User and implementing the `org_attribute_path` is not required.

### PR DESCRIPTION
**Pull request by kalleep**
**What is this feature?**
This PR adds the org role mapping feature to the Gitlab connector.

**Which issue(s) does this PR fix?**:
Fixes #88620

**Special notes for your reviewer:**

This is the GitLab part of https://github.com/grafana/grafana/pull/77683.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 1]
```diff
@@ -149,15 +149,28 @@ To map the server administrator role, use the `allow_assign_grafana_admin` confi
...
 
 To ease configuration of a proper JMESPath expression, go to [JMESPath](http://jmespath.org/) to test and evaluate expressions with custom payloads.
 
 ### Role mapping examples
 
 This section includes examples of JMESPath expressions used for role mapping.
 
+##### Org roles mapping example
+
+The Gitlab integration uses the external users' groups in the `org_mapping` configuration to map organizations and roles based on their Gitlab group membership.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:162]:
```suggestion
The GitLab integration uses the external users' groups in the `org_mapping` configuration to map organizations and roles based on their GitLab group membership.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 2]
```diff
@@ -149,15 +149,28 @@ To map the server administrator role, use the `allow_assign_grafana_admin` confi
...
+
+In this example, the user has been granted the role of a `Viewer` in the `org_foo` organization, and the role of an `Editor` in the `org_bar` and `org_baz` orgs.
+
+The external user is part of the following GitHub teams: `groupd-1` and `group-2`.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:166]: Should "GitHub" be "GitLab"?

**Review by chri2547 — COMMENTED** [review]
A couple of nits, otherwise looks good! Thank you for the contribution!

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md (hunk 2 above)
Comment (kalleep) [reply docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:166]: Yes you are correct, thanks

**Review by chri2547 — COMMENTED** [review]
Docs approved. I'll let others give the formal approval required for merge. Thank you.

**Review by mgyongyosi — APPROVED** [review]

**Pull request merged**